mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -868,12 +872,23 @@ fn try_parse_generic_error(request_result: &MetaRequestResult) -> Option<S3Reque
         }
     }
 
+    /// Try to look for error related to no signing credentials
+    fn try_parse_no_credentials(request_result: &MetaRequestResult) -> Option<S3RequestError> {
+        let crt_error_code = request_result.crt_error.raw_error();
+        if crt_error_code == 6146 {
```

Comment:
Add a comment for this number that it's `AWS_AUTH_SIGNING_NO_CREDENTIALS` from the CRT. Would be great if we can use enum instead of raw number. I think `bindgen` already generated these enum for us, what we have to do is creating a wrapper for them in `mountpoint-s3-crt`.

Anyway, not blocking on that since we only want this specific error code. Adding a comment is fine for me.

Reply:
Agree we can do this in a separate change. I think the right thing would be for the `mountpoint_s3_crt::common::error` to export all the raw error codes from the bindings, maybe in a `raw` submodule. But it gets a bit weird because of how bindgen converts enums (what integer type to use), so will need some fiddling around.

Reply:
yes, there is `pub enum aws_auth_errors`. For now, I am only adding the comments here.